### PR TITLE
Openstack4J integration

### PR DESCRIPTION
**Pull request by pawelpros**
Removed support for jcloud and updated code to use openstack4j library with keystone version 3.

Additionally:
-     Updated floating ip behavior
-     updated deprecated usage of extensions to proper supported ones
-     updated OSGI
-     removed all dependencies connected with jcloud
-     added dependencies to openstack4j

### CODE REVIEW

File: osc-uber-jcloud/README.md [hunk 1]
```diff
@@ -1,7 +0,0 @@
-# Uber jcloud
```
Comment (arvindn05) [inline osc-uber-jcloud/README.md]: can you add the appropritate readme?

File: osc-server/src/main/java/org/osc/core/broker/util/VirtualizationConnectorUtil.java [hunk 2]
```diff
@@ -110,25 +110,20 @@ private void delayedInit() {
...
 
-                    Endpoint endPoint = new Endpoint(vcDto.getProviderIP(), vcDto.getAdminTenantName(),
+                    Endpoint endPoint = new Endpoint(vcDto.getProviderIP(), vcDto.getAdminDomainId(), vcDto.getAdminTenantName(),
                             vcDto.getProviderUser(), vcDto.getProviderPassword(), isHttps,
                             SslContextProvider.getInstance().getSSLContext());
-                    keystoneAPi = new JCloudKeyStone(endPoint);
-                    keystoneAPi.listTenants();
 
+                    Openstack4jKeystone keystoneAPi = new Openstack4jKeystone(endPoint);
+                    keystoneAPi.listProjects();
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/util/VirtualizationConnectorUtil.java]: need to close keystone

File: osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/ValidateSecurityGroupTenantTask.java [hunk 3]
```diff
@@ -55,25 +54,19 @@ public void executeTransaction(EntityManager em) throws Exception {
...
-                // Tenant was deleted, mark Security Group for deleting as well
-                OSCEntityManager.markDeleted(em, this.securityGroup, this.txBroadcastUtil);
-            } else {
-                // Sync the tenant name if needed
-                if (!tenant.getName().equals(this.securityGroup.getTenantName())) {
-                    this.log.info("Security Group tenant name updated from " + this.securityGroup.getTenantName() + " to " + tenant.getName());
-                    this.securityGroup.setTenantName(tenant.getName());
-                    OSCEntityManager.update(em, this.securityGroup, this.txBroadcastUtil);
-                }
+        Openstack4jKeystone keystone = new Openstack4jKeystone(new Endpoint(this.securityGroup.getVirtualizationConnector()));
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/ValidateSecurityGroupTenantTask.java]: need to close keystone...please recheck all instances so we dont leave stale resources

File: osc-uber-jcloud/README.md (hunk 1 above)
Comment (pawelpros) [inline osc-uber-jcloud/README.md]: Ok, I'll add it

File: osc-server/src/main/java/org/osc/core/broker/util/VirtualizationConnectorUtil.java (hunk 2 above)
Comment (pawelpros) [inline osc-server/src/main/java/org/osc/core/broker/util/VirtualizationConnectorUtil.java]: Updated

**Comment by arvindn05** [comment]
closing as #367 continues this work

**Pull request closed without merging**